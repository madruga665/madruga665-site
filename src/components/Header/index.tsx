import Brand from "../Brand";
import NavLinks from "../Navlinks";
import styles from "./styles.module.scss";

const Header = () => {
  return (
    <header className={styles.Header}>
      <Brand />
      <NavLinks />
    </header>
  );
};

export default Header;
