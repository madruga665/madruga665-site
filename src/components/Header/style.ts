import styled from "styled-components";
import { Colors } from "../../styles/variables";

export const HeaderStyled = styled("header")`
  display: flex;
  flex-direction: column;
  align-items: center;
  position: fixed;
  padding: 0.5rem 0;
  gap: 2.5rem;
  width: 100%;
  height: 8rem;
  display: flex;
  background: ${Colors.midnightBlue};
  color: ${Colors.clouds};

  @media only screen and (min-width: 768px) {
    flex-direction: row;
    justify-content: space-between;
    height: 5rem;
    padding: 0.5rem 1rem;
  }
`;
