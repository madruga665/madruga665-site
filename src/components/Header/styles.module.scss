@import '../../styles/colors.module.scss';

.Header {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 0.5rem 0;
  gap: 2.5rem;
  width: 100%;
  height: 8rem;
  background: $color-midnightBlue;
  color: $color-clouds;

  @media only screen and (min-width: 768px) {
    flex-direction: row;
    justify-content: space-between;
    height: 5rem;
    padding: 0 1rem;
  }
}