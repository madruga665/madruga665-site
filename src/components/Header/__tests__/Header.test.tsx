import { render, screen } from "@testing-library/react";
import Header from "..";
import { MemoryRouter } from "react-router-dom";

describe("Testes do componente Header", () => {
  const route = "/portifolio/home";

  test("Snapshot do componente", () => {
    const { container } = render(
      <MemoryRouter initialEntries={[route]}>
        <Header />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });

  test("O logo esta sendo renderizado", () => {
    render(
      <MemoryRouter initialEntries={[route]}>
        <Header />
      </MemoryRouter>
    );
    const madrugaLogo = screen.getByRole("img", {
      name: /logo de madruga/i,
    });
    expect(madrugaLogo).toBeInTheDocument();
  });

  test("O nome está sendo renderizado", () => {
    render(
      <MemoryRouter initialEntries={[route]}>
        <Header />
      </MemoryRouter>
    );
    const madrugaName = screen.getByText(/<uciano mâncio\/>/i);
    expect(madrugaName).toBeInTheDocument();
  });

  test("A navBar está sendo renderizada", () => {
    render(
      <MemoryRouter initialEntries={[route]}>
        <Header />
      </MemoryRouter>
    );
    const navBar = screen.getByRole("navigation");
    expect(navBar).toBeInTheDocument();
  });
});
