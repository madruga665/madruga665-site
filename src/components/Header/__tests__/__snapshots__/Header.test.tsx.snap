// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testes do componente Header Snapshot do componente 1`] = `
<div>
  <header
    class="sc-dkPtRN bxxqEN"
  >
    <div
      class="sc-bdvvtL klUASx"
    >
      <img
        alt="Logo de Madruga"
        height="50"
        src="/assets/icons8-circled-m-80.png"
        width="50"
      />
      <h2>
        &lt;
        <span>
          L
        </span>
        uciano 
        <span>
          A
        </span>
        m√¢ncio
        /&gt;
      </h2>
    </div>
    <nav
      class="sc-gsDKAQ gkFwAu"
    >
      <ul>
        <li>
          <a
            href="/portifolio"
          >
            HOME
          </a>
        </li>
        <li>
          <a
            href="/portifolio/projetos"
          >
            PROJETOS
          </a>
        </li>
        <li>
          <a
            href="/portifolio/contato"
          >
            CONTATO
          </a>
        </li>
        <li>
          <a
            href="/portifolio/sobre"
          >
            SOBRE
          </a>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
